app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt

```diff
@@ -106,31 +100,105 @@ class LinkPreviewDialog : ExtendedBottomSheetDialogFragment(), LinkPreviewErrorV
                 show()
             }
         }
-        L10nUtil.setConditionalLayoutDirection(binding.root, pageTitle.wikiSite.languageCode)
-        loadContent()
-        funnel = LinkPreviewFunnel(WikipediaApp.instance, historyEntry.source)
+        L10nUtil.setConditionalLayoutDirection(binding.root, viewModel.pageTitle.wikiSite.languageCode)
+        renderViewStates()
+//        viewModel.loadContent()
+        funnel = LinkPreviewFunnel(WikipediaApp.instance, viewModel.historyEntry.source)
```

Comment:
Let's move this creation of the `funnel` object up _before_ the call to `renderViewStates()` function. Currently there can be a state where a uiState is reached before `funnel` is initialized.